homeassistant/components/binary_sensor/mystrom.py

```diff
@@ -0,0 +1,98 @@
+"""
+Support for the myStrom buttons.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.mystrom/
+"""
+import asyncio
+import logging
+
+from homeassistant.const import HTTP_UNPROCESSABLE_ENTITY
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.components.binary_sensor import (BinarySensorDevice, DOMAIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['http']
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up myStrom Binary Sensor."""
+    button = None
+
+    hass.http.register_view(MyStromView(button, async_add_devices))
+
+    return True
+
+
+class MyStromView(HomeAssistantView):
+    """View to handle requests from myStrom buttons."""
+
+    url = '/api/mystrom'
+    name = 'api:mystrom'
+
+    def __init__(self, button, add_devices):
+        """Initialize the myStrom URL endpoint."""
+        self.button = button
+        self.add_devices = add_devices
+
+    @asyncio.coroutine
+    def get(self, request):
+        """The GET request received from a myStrom button."""
+        res = yield from self._handle(request.app['hass'], request.GET)
+        return res
+
+    @asyncio.coroutine
+    def _handle(self, hass, data):
+        """Handle requests to the myStrom endpoint."""
+        button_action = list(data.keys())[0]
+        button_id = data[button_action]
+        entity_id = '{}.{}_{}'.format(DOMAIN, button_id, button_action)
+        state = hass.states.get(entity_id)
+
+        if button_action in ['single', 'double', 'long', 'touch']:
+            if state is None:
+                _LOGGER.info("New myStrom button/action detected: %s, %s",
+                             button_id, button_action)
+                self.button = MyStromBinarySensor(
+                    '{}_{}'.format(button_id, button_action))
+                hass.async_add_job(self.add_devices, [self.button])
+            else:
+                new_state = 'on' if state.state == 'off' else 'off'
+                hass.states.async_set(entity_id, new_state, state.attributes)
```

Comment:
We should not call `states.async_set` directly but instead push our changes to the entity.